pkg/reconciler/trigger/trigger.go

```diff
@@ -66,12 +64,13 @@ const (
 type Reconciler struct {
 	*reconciler.Base
 
-	triggerLister            listers.TriggerLister
-	subscriptionLister       messaginglisters.SubscriptionLister
-	brokerLister             listers.BrokerLister
-	serviceLister            corev1listers.ServiceLister
-	resourceTracker          duck.ResourceTracker
-	kresourceInformerFactory apisduck.InformerFactory
+	triggerLister      listers.TriggerLister
+	subscriptionLister messaginglisters.SubscriptionLister
+	brokerLister       listers.BrokerLister
+	serviceLister      corev1listers.ServiceLister
+	tracker            tracker.Interface
```

Comment:
This needs a big comment on it saying exactly what can be tracked with this tracker (just Brokers?). And what needs to be tracked by the other two.

If you want to be super paranoid, make a struct that only exposes `TrackBroker`, `TrackDeployment`, etc. that can actually be tracked by this thing. Then we can use type safety to enforce it is only tracking what it should.

Reply:
There were some comments in the controller.go. Moved them here, you are right.
I think we can do some follow up(s) with the type safety, and other improvement later on, if that's ok with you